x-pack/plugins/reporting/export_types/csv_from_savedobject/server/lib/generate_csv_search.ts

```diff
@@ -0,0 +1,211 @@
+    timeFilter = null;
+    timezone = null;
+  }
+
+  const combinedFilter = timeFilter ? [timeFilter].concat(searchSourceFilter) : searchSourceFilter;
+
+  return { combinedFilter, includes, timezone };
+};
+
+interface IndexPatternField {
+  scripted: boolean;
+  lang?: string;
+  script?: string;
+  name: string;
+}
+
+export async function generateCsvSearch(
+  req: Request,
+  server: KbnServer,
+  logger: Logger,
+  searchPanel: SearchPanel
+): Promise<CsvResultFromSearch> {
+  const { savedObjects, uiSettingsServiceFactory, fieldFormatServiceFactory } = server;
+  const savedObjectsClient = savedObjects.getScopedSavedObjectsClient(req);
+  const { indexPatternSavedObjectId, timerange } = searchPanel;
+  const savedSearchObjectAttr = searchPanel.attributes as SavedSearchObjectAttributes;
+  const { indexPatternSavedObject } = await getDataSource(
+    savedObjectsClient,
+    indexPatternSavedObjectId
+  );
+  const uiConfig = uiSettingsServiceFactory({ savedObjectsClient });
+
+  const kibanaSavedObjectMeta = savedSearchObjectAttr.kibanaSavedObjectMeta;
+  const { searchSource } = kibanaSavedObjectMeta as { searchSource: SearchSource };
+
+  const { filter: searchSourceFilter, query: searchSourceQuery } = searchSource;
+  const { combinedFilter, includes, timezone } = getTimebasedParts(
+    indexPatternSavedObject,
+    timerange,
+    savedSearchObjectAttr,
+    searchSourceFilter
+  );
+  const esQueryConfig = await getEsQueryConfig(uiConfig);
+  const [sortField, sortOrder] = savedSearchObjectAttr.sort;
+
+  const scriptFields = indexPatternSavedObject.fields
+    .filter((f: IndexPatternField) => f.scripted)
+    .reduce((accum: any, curr: IndexPatternField) => {
+      return {
+        ...accum,
+        [curr.name]: {
+          script: {
+            source: curr.script,
+            lang: curr.lang,
+          },
+        },
+      };
+    }, {});
+  const searchRequest: SearchRequest = {
+    index: indexPatternSavedObject.title,
+    body: {
+      _source: { includes },
+      docvalue_fields: [],
+      query: buildEsQuery(
+        indexPatternSavedObject,
+        searchSourceQuery,
+        combinedFilter,
+        esQueryConfig
+      ),
+      script_fields: scriptFields,
+      sort: [{ [sortField]: { order: sortOrder } }],
+    },
+  };
+
+  const { callWithRequest } = server.plugins.elasticsearch.getCluster('data');
+  const callCluster = (...params: any[]) => callWithRequest(req, ...params);
+  const config = server.config();
+  const [formatsMap, uiSettings] = await Promise.all([
+    fieldFormatServiceFactory(uiConfig).then((fieldFormats: any) => {
+      return fieldFormatMapFactory(indexPatternSavedObject, fieldFormats);
+    }),
+    getUiSettings(uiConfig),
+  ]);
+
+  const generateCsvParams: GenerateCsvParams = {
+    searchRequest,
+    callEndpoint: callCluster,
+    fields: includes,
+    formatsMap,
+    metaFields: [],
+    conflictedTypesFields: [],
+    cancellationToken: [],
```

Comment:
If the search is taking longer than the reporting timeout config, it should be automatically cancelled. But this might be more of a TODO for a later phase. Thoughts?

Reply:
Yeah, I don't think we should short-circuit/bypass that

Reply:
This was implemented in a later PR, closing!